backend/src/models/reports/Tab_23_rpt_PA_Fiscal_Registry.js

```diff
@@ -1,16 +1,147 @@
-//const dbConnection = require("@database/databaseConnection");
-//const { knex } = dbConnection();
+// Libs
+const { knex } = require("@database/databaseConnection")();
+const { keyBy, map } = require("lodash");
 
-const getReport = (portfolio) => {
-  const query = "removeme"; //knex().raw();
-  return query;
-};
+// Utilities
+const { groupByProperty } = require("./helpers");
 
-module.exports = {
-  required: [],
-  getAll: async (query) => {
-    const { portfolio } = query;
-    const [report] = await Promise.all([getReport(portfolio)]);
-    return { report };
+/**
+ * Retrieves the data for various financial metrics based on the fiscal year.
+ *
+ * @param   {number | string | Array} Parameter- The fiscal, Date, or Portfolio(s) to grab data for
+ * @returns {Promise}                            - A promise that resolves to the query result
+ */
+const queries = {
+  // The columns on which to calculate totals.
+  columns: {
+    total: "report.planned_budget",
   },
+
+  // The query to get the fiscal year.
+  fiscal: (fiscal) =>
+    knex("fiscal_year").select("fiscal_year").where("fiscal_year.id", fiscal).first(),
+
+  report: (fiscal) =>
+    knex
+      .select({
+        portfolio_name: "po.portfolio_name",
+        project_number: "p.project_number",
+        project_name: "p.project_name",
+        project_manager: knex.raw(`(
+          SELECT first_name || ' ' || last_name
+          FROM contact
+          WHERE id = p.project_manager
+        )`),
+        start_date: knex.raw(`to_char(p.planned_start_date, 'dd-Mon-yy')`),
```

Comment:
i thought we had a variable that we placed in for the date format, i worry about hard coding this kind of thing

Reply:
believe it or not, this function was in Gary's query:

https://apps.itsm.gov.bc.ca/bitbucket/projects/DES/repos/pmo-postgres-queries/browse/rpt_PA_Fiscal_Registry.md

Reply:
yes i believe it, but it still doesn't mean we should leave it. but i suspect it is in many other queries.
I'm pretty sure there was a variable from something like the utility or helper that defines this

Reply:
I'll take a look

is this what you mean?

`knex.raw(`TO_CHAR(pd.start_date :: DATE, '${dateFormat}')`)`

It's one way to at least have control over the format of the date returned in a query